services/QuillLMS/app/queries/prompt_feedback_history.rb

```diff
@@ -0,0 +1,54 @@
+class PromptFeedbackHistory
+
+    def self.run(activity_id)
+        promptwise_postprocessing promptwise_sessions(activity_id)
+    end
+
+    def self.promptwise_sessions(activity_id)
+        activity_sessions = ActivitySession.where(activity_id: activity_id).includes(:feedback_sessions)
+        feedback_session_uids = activity_sessions.reduce([]) {|memo, a_s| memo.concat a_s.feedback_sessions.pluck(:uid)}
+
+        sql = <<~SQL 
+          SELECT prompt_id, feedback_session_uid, count(*) as session_count, bool_or(optimal) as at_least_one_optimal, MAX(attempt) as attempt_cardinal
+          FROM feedback_histories
+          WHERE feedback_session_uid IN (#{feedback_session_uids.map{|e| "'#{e}'"}.join(',')})
+          GROUP BY prompt_id, feedback_session_uid
+        SQL
+        FeedbackHistory.find_by_sql(sql)
+    end
+
+    def self.promptwise_postprocessing(grouped_feedback_histories)
+        prompt_hash = {}
+
+        prompts = Comprehension::Prompt.where(id: grouped_feedback_histories.map(&:prompt_id))
+   
+        grouped_feedback_histories.each do |prompt_session|
+            prompt_id = prompt_session.prompt_id
+
+            prompt_hash[prompt_id] ||= {
+                optimal_final_attempts: 0, 
```

Comment:
I think we also need a `not_optimal_final_attempts` attribute here, since it looks like the spec calls for both a percentage and an actual number.

Reply:
You are correct, but as I mentioned above:

> Note: this PR implements the MVP version of the Notion task; not all fields are implemented. The implemented fields are designated via checkmarks in Notion.

Reply:
Yeah, I see that, but I'm not sure what the point of omitting the `not_optimal_final_attempts` is here when we have the `optimal_final_attempts` and the percentages for both. Since these are all going in the same table cell, it feels like it would make sense to also include that in the MVP.

Reply:
Which is to say, you have that box checked off in the spec but it isn't in this query.